.github/workflows/eks-e2e-test.yml

```diff
@@ -0,0 +1,160 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: MIT
+
+name: Reusable EKS E2E Test
+
+env:
+  TERRAFORM_AWS_ASSUME_ROLE_DURATION: 14400 # 4 hours
+
+on:
+  workflow_call:
+    inputs:
+      terraform_dir:
+        required: true
+        type: string
+      job_id:
+        required: true
+        type: string
+      test_props:
+        required: true
+        type: string
+      test_repo_name:
+        required: true
+        type: string
+      test_repo_url:
+        required: true
+        type: string
+      test_repo_branch:
+        required: true
+        type: string
+      cloudwatch_agent_repository:
+        required: true
+        type: string
+      cloudwatch_agent_tag:
+        required: true
+        type: string
+      cloudwatch_agent_operator_repository:
+        required: true
+        type: string
+      cloudwatch_agent_target_allocator_repository:
+        required: false
+        type: string
+      cloudwatch_agent_operator_tag:
+        required: true
+        type: string
+      region:
+        required: true
+        type: string
+      helm_charts_branch:
+        required: true
+        type: string
+      terraform_assume_role:
+        required: true
+        type: string
+      agent_config:
+        required: true
+        type: string
+      prometheus_config:
+        required: false
+        type: string
+        default: ""
+      otel_config:
+        required: false
+        type: string
+        default: ""
+      sample_app:
+        required: true
+        type: string
+
+jobs:
+  EKSE2ETest:
+    name: 'EKSE2ETest'
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        arrays: ${{ fromJson(inputs.test_props) }}
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          repository: ${{inputs.test_repo_name}}
+          ref: ${{inputs.test_repo_branch}}
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: ${{ inputs.terraform_assume_role }}
+          aws-region: ${{ inputs.region }}
+          role-duration-seconds: ${{ env.TERRAFORM_AWS_ASSUME_ROLE_DURATION }}
+
+      - name: Cache if success
+        id: cache_if_success
+        uses: actions/cache@v3
+        with:
+          path: go.mod
+          key: ${{inputs.region}}-${{ github.sha }}-${{ matrix.arrays.os }}-${{ matrix.arrays.arc }}-${{ matrix.arrays.test_dir }}
+
+      - name: Login ECR
```

Comment:
If we move this to terraform we can use this workflow for other environments like ECS

Reply:
Looking at this, we should be able to refactor it as is for ECS.